### ISSUE TITLE
Middleware loggers unaware about rescue_from handlers

### ISSUE THREAD

**Issue by thedarkside** [ruby-grape/grape#1734]
As far as i digged in the code it seems that loggers realized as middleware are not aware of changes to exceptions made by rescue_from handlers.

I stumbled upon this as i saw log entries with a wrong statuscode in my application. I am catching an UnauthorizedException in a rescue_from block and setting the desired statuscode with `error! e, 401`. It works as expected but produces a 500 log entry.

Maybe worth a discussion about possible solutions?

**Comment by dblock** [comment]
Care to provide a more detailed example with some code maybe?

**Comment by thedarkside** [comment]
@dblock thank you for listening but i think i figured it out myself. I've inserted now the logger before Grape::Middleware::Error which handles the rescue_from blocks and then passes the exception with the right status code to the logger.

It was tricky to figure out the actual middleware stack. I just saw it in a stacktrace while debugging. It would be nice to have a convinient rake task which prints out the whole middleware stack just like in rails + some more detailed documentation about it.
